src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -3137,6 +3137,55 @@ int32_t NaClSysImcSocketPair(struct NaClAppThread *natp,
   return retval;
 }
 
+int32_t NaClSysSocketPair(struct NaClAppThread *natp,
+                          int                  domain,
+                          int                  type,
+                          int                  protocol,
+                          int                  *fds) {
+
+  struct NaClApp          *nap = natp->nap;
+  void                    *hd_struct = NULL;
+  struct NaClHostDesc     *hd;
+  int                     lind_fd;
+  int                     nacl_fd;
+  int                     user_fd;
+  int32_t                 retval;
+
+  NaClLog(2, "Cage %d Entered NaClSysSocketPair(0x%08"NACL_PRIxPTR", "
+           "%d, %d, %d, 0x%08"NACL_PRIx32")\n",
+           nap->cage_id, (uintptr_t)natp, domain, type, protocol, (uintptr_t)fds);
+
+  //Preprocessing start
+  hd_struct = malloc(sizeof(struct NaClHostDesc)*2);
+  if (!hd_struct) {
+    retval = -NACL_ABI_ENOMEM;
+    return retval;
+  }
+  //Preprocessing end
```

Comment:
The *processing end/start comments are kind of clutter

Reply:
I think you are right, will remove them.